### ISSUE TITLE
a list of all dances

### ISSUE THREAD

**Issue by jmdyck** [contradb/contra#403]
I'm thinking of adding contradb.com as a source in [my index](http://www.ibiblio.org/contradance/index/). Is there an URL I can hit that will return a list of all (publicly visible) dances in the db?

**Comment by dcmorse** [comment]
The root page is meant to communicate all the dances. Does the fact that it's paginated bother you?

**Comment by jmdyck** [comment]
That, plus the fact that if I fetch that URL (using wget, say), the response doesn't actually list any dances -- the table of dances only gets filled in via javascript.

**Comment by dcmorse** [comment]
Are you scraping? So is what you really want a web api for fetching dances? That speaks JSON?

**Comment by jmdyck** [comment]
Yup, I'm scraping. I wouldn't say that a web API that speaks JSON is what I "really want", but if you provided it, I believe it would fit my purpose.

**Comment by dcmorse** [comment]
What do you really want?

<table>
<row>
<th>Dance Title</th>
<th>Choreographer</th>
<th>Formation</th>
<th>Hook</th>
</row>
...rows...
</table>

**Comment by jmdyck** [comment]
Title, Choreographer, Formation, and dance id/number.

**Comment by dcmorse** [comment]
Have a look at the output of this:
```
curl https://contradb.com/api/v1/dances -X POST -H "Content-Type: application/json" -d '{"count": "2", "offset": "30", "filter": ["publish", "all"] }' 
curl https://contradb.com/api/v1/dances -X POST -H "Content-Type: application/json" -d '{"count": "2", "offset": "30", "filter": ["compare", ["constant", "0"], "<", ["tag", "verified"]]}'
```

- `count` is the maximum number of dances to fetch - right now there's no limit. 
- `offset` is the number of dances to skip past before enumerating the rest
- `filter` tells it what types of dances to get. The `["pubish": "all"]` filter gets dances that the dance-transcribers think are ready to call (about 1000 dances). The `["compare", ["constant", "0"], "<", ["tag", "verified"]]` one gets dances that logged-in users say that they actually have called, in a hall, with people (about 200 dances for the moment).
